octree/include/pcl/octree/octree_iterator.h

```diff
@@ -384,11 +362,17 @@ namespace pcl
         explicit
         OctreeDepthFirstIterator (OctreeT* octree_arg, unsigned int max_depth_arg = 0);
 
-        /** \brief Empty deconstructor. */
-        virtual
-        ~OctreeDepthFirstIterator ();
+        /** \brief Copy Constructor.
+         * \param[in] other Another OctreeDepthFirstIterator to copy from
+         */
+        OctreeDepthFirstIterator (const OctreeDepthFirstIterator& other)
+          : OctreeIteratorBase<OctreeT> (other)
+          , stack_ (other.stack_)
+        {
+          this->current_state_ = stack_.size()? &stack_.back() : NULL;
```

Comment:
Style

Also, can you please explain why we inherit the state from other, i.e. `this->current_state_ = other.current_state_`?

Reply:
I'm gonna redirect you to the explanation I gave @frozar when I had a first look at the copy constructor fix he did. Let me know in case you need further clarification. Here's the link https://github.com/PointCloudLibrary/pcl/pull/1983#discussion_r154152023

The important point to remember is that `current_state_` points to an address, which is normally the back element of the stack or the front of the FIFO vector (depending on the type of iterator) and when you perform the copy, this pointer needs to be updated the equivalent element of the stack/fifo container your object has. Also notice that `current_state_` is not a direct pointer to an octree node. Hence it can't be simply copied.

Reply:
Thanks, makes sense.